client/src/components/Citation/services.ts

```diff
@@ -1,13 +1,14 @@
 import axios from "axios";
-import { getAppRoot } from "onload/loadConfig";
-import { rethrowSimple } from "utils/simple-error";
 
-export async function getCitations(source, id) {
+import { getAppRoot } from "@/onload/loadConfig";
+import { rethrowSimple } from "@/utils/simple-error";
+
+export async function getCitations(source: string, id: string) {
```

Comment:
I guess you can create an interface called `Catation` with the properties you know will be present, similar to what you defined in the object literal here:

https://github.com/galaxyproject/galaxy/pull/17483/files#diff-4656fc7d33ea8ddd059a41d462e492f5f1957fbb84a07b900888c3d4d7353b20R10-R24

Something like:
```ts
export interface Citation {
    raw: string;
    cite: {
        data: {
            URL: string;
        }[];
        format: (
            format: string,
            options: {
                format: string;
                template: string;
                lang: string;
            }
        ) => string;
    };
}
```

And then:

```suggestion
export async function getCitations(source: string, id: string): Promise<Citation[]>  {
```

And finally you can reuse that interface in the components :)

Reply:
Thanks for the constructive feedback! I applied the changes :)